python_modules/dagster/dagster/core/asset_defs/materialize.py

```diff
@@ -0,0 +1,44 @@
+from typing import Any, List, Mapping, Optional, Sequence
+
+import dagster._check as check
+
+from ..definitions import ResourceDefinition
+from ..execution.execute_in_process_result import ExecuteInProcessResult
+from ..execution.with_resources import with_resources
+from ..instance import DagsterInstance
+from ..storage.fs_io_manager import fs_io_manager
+from .assets import AssetsDefinition
+from .assets_job import build_assets_job
+from .source_asset import SourceAsset
+
+
+def materialize(
+    assets: Sequence[AssetsDefinition],
+    source_assets: Optional[Sequence[SourceAsset]] = None,
+    run_config: Any = None,
+    instance: Optional[DagsterInstance] = None,
+) -> ExecuteInProcessResult:
+    """
+    Executes an in-process run which materializes assets that fulfill selection.
```

Comment:
I think it would be good to mention in this docstring that this method will write asset outputs to the filesystem (via `fs_io_manager`).

Reply:
I think also worth mentioning that the materializations will happen serially in a single thread.